server/app.py

```diff
@@ -1,33 +1,90 @@
+app.register_blueprint(home_handler)
+app.register_blueprint(ping_handler)
 
+@login.user_loader
+def load_user(id):
+    return User.objects(username=id).first()
 
+class User(db.Document, UserMixin):
+    username = db.StringField(max_length=20)
+    email = db.EmailField(max_length=60)
+    password = db.StringField(max_length=20)
+    password_hash = db.StringField(max_length=128)
 
+    def set_password(self, password):
+        self.password_hash = generate_password_hash(password)
+
+    def check_password(self, password):
+        return check_password_hash(self.password_hash, password)
+
+    def __repr__(self):
+        return '<User {}>'.format(self.username)
 
-app.register_blueprint(home_handler)
-app.register_blueprint(ping_handler)
 
-@app.route("/", methods=["GET"])
+class LoginForm(FlaskForm):
+    username = StringField('Username', validators=[DataRequired()])
+    password = PasswordField('Password', validators=[DataRequired()])
+    submit = SubmitField('Sign In')
+
+class SignupForm(FlaskForm):
+    username = StringField('Username', validators=[DataRequired(), Length(min=6)])
+    email = EmailField('Email', validators=[DataRequired(), Length(min=6)])
+    password = PasswordField('Password', validators=[DataRequired(), Length(min=7), EqualTo('confirm', message='Passwords must match')])
+    confirm = PasswordField('Confirm')
+    submit = SubmitField('Submit')
+
+    def validate_username(self, username):
+        user = User.objects(username=username.data).first()
+        if user is not None:
+            raise ValidationError('Please use a different username.')
+
+    def validate_email(self, email):
+        user = User.objects(email=email.data).first()
+        if user is not None:
+            raise ValidationError('Please use a different email address.')
+
+
+@app.route('/')
 def index():
-  return f"{os.getenv('TEST_ENV')}... App is running..."
+    return render_template('index.html', title='home')
+
+@login_required
+@app.route('/login', methods = ['GET','POST'])
+def login():
+    if current_user.is_authenticated:
+        return redirect(url_for('index'))
+    form = LoginForm()
+    if form.validate_on_submit():
+        user = User.objects(username=form.username.data).first()
+        if user is None or not user.check_password(form.password.data):
+            return redirect(url_for('login')), 401
+        login_user(user)
+        return redirect(url_for('index'))
+    return render_template('login.html', title='Sign In', form=form)
 
-if __name__ == "__main__":
-  app.run(port=5000, debug=True)
+@app.route('/signup', methods = ['GET','POST'])
+def signup():
+    form = SignupForm()
+    if form.validate_on_submit():
+        new_user = User(username=form.username.data, email=form.email.data)
+        new_user.set_password(form.password.data)
+        new_user.save()
+        return redirect(url_for('login')), 201
+    return render_template('signup.html', title='Sign Up', form=form)
```

Comment:
In this app we're using a React frontend and a Python JSON API backend. This means that all UI (including forms) will be coded in React. Your [ticket](https://app.zenhub.com/workspaces/zen-team-blt-60201e4aa8b22b00124a745f/issues/hatchways/team-blt/8) is for the backend only. In your routes, rather than render an html template, you will receive a JSON object containing the authentication details and will return a JSON response.

Reply:
Hi Aeciorc. I have corrected the code as you requested. I will now open a new pr.